### ISSUE TITLE
Doomguards (maybe other pets) are not affected by bloodlust

### ISSUE THREAD

**Issue by BalanCV** [gamefreedomgit/Maelstrom#3823]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**
"Pets and controlled guardians (e.g. druid Treants) can be affected by Bloodlust/Heroism/Warp Time/Ancient hysteria auras.
gamefreedomgit/Maelstrom#3735 "
This simply does not apply to doomguard right now.
I have tested treats and those do get the BL buff.
There might also be other pets or guardians not benefiting from it either.

Personal opinion:
Since currently it is not working at all, the way a fix is implemented is an opportunity to enhance gameplay.
You should make these guardians inherit the bloodlust(only) % haste instead of gaining the bloodlust buff after being summoned.

Logic behind opinion:
Warlock and other classes have a high-level gameplay centered around Snapshotting. Doomguard is the biggest damage cooldown. IF bloodlust is applied only IF pet is out during BL it forces the warlock to ignore other trinket procs in order to rush summoning the "naked" doomguard so it receives BL since many encounters do BL on pull OR force your team to BL twice for pets only ( if you even have 2 BL capable people in group )
Basically by letting guardians/pets inherit ONLY the bloodlust buff  from owner you would allow a larger snapshot window making for a more smooth gameplay while having literally the SAME effect as pets getting BL buff after being summoned. (probably an easier fix as well)

**How to reproduce:**
0 personal haste image BEFORE bloodlust cast time 3.0 https://imgur.com/a/rSeTG8d
0 personal haste image After bloodlust same cast time 3.0 https://imgur.com/a/gbs6LYz

**How it should work:**

**Database links:**

**Comment by Sketish** [comment]
Doomguard doesnt have control panel like treants or wolves so he shouldnt be affected by bloodlust
